custom_components/tuya_local/devices/nac_ap470_purifier.yaml

```diff
@@ -0,0 +1,146 @@
+name: Air purifier
+products:
+  - id: y7iiubdc5sufjvo3
+    name: "NAC AP470 Air Purifier"
+primary_entity:
+  entity: fan
+  translation_key: fan_with_presets
+  icon: "mdi:air-purifier"
+  dps:
+    - id: 1
+      type: boolean
+      name: switch
+    - id: 4
+      type: string
+      name: preset_mode
+      mapping:
+        - dps_val: 1
+          value: Auto
+        - dps_val: 2
+          value: Sleep
+        - dps_val: 3
+          value: Level 1
+        - dps_val: 4
+          value: Level 2
+        - dps_val: 4
+          value: Level 3
+        - dps_val: 6
+          value: Level 4
+    - id: 21
+      name: fault_code
+      type: bitfield
+secondary_entities:
+  - entity: sensor
+    class: pm25
+    dps:
+      - id: 2
+        type: integer
+        name: sensor
+        class: measurement
+        unit: ugm3
+  - entity: sensor
+    name: Filter left days
+    icon: "mdi:air-filter"
+    category: diagnostic
```

Comment:
`class: duration` might apply here (I'm not sure it changes anything beyond hours though)